### PR TITLE
Dealing with standard units.

### PR DESCRIPTION
**Pull request by hsorby**
This PR looks at defining the way units are handled that are standard units.  The two tests show the new behaviour.

Addresses issue #545.

### CODE REVIEW

**Comment by kerimoyle** [comment]
See also #525 and pr #526?

File: src/printer.cpp [hunk 1]
```diff
@@ -200,6 +200,8 @@ std::string Printer::PrinterImpl::printUnits(const UnitsPtr &units) const
             repr += " id=\"" + units->id() + "\"";
         }
         repr += "/></import>";
+    } else if (isStandardUnit(units)) {
+        // Do nothing
```
Comment (agarny) [inline src/printer.cpp]: Missing full stop at the end of the comment.

File: src/utilities.cpp [hunk 2]
```diff
@@ -431,6 +432,15 @@ bool isStandardUnitName(const std::string &name)
     return standardUnitsList.count(name) != 0;
 }
 
+bool isStandardUnit(const UnitsPtr &units)
+{
+    bool is = false;
+    if (units != nullptr) {
+        is = units->isBaseUnit() && isStandardUnitName(units->name());
+    }
+    return is; //standardUnitsList.count(name) != 0;
```
Comment (agarny) [inline src/utilities.cpp]: Why that comment? I would delete it.

Having said that, I would rewrite the whole method to simply read:

```
bool isStandardUnit(const UnitsPtr &units)
{
    if ((units != nullptr) && units->isBaseUnit() && isStandardUnitName(units->name())) {
        return true;
    }

    return false;
}
```

File: src/utilities.h [hunk 3]
```diff
@@ -390,6 +390,8 @@ void removeComponentFromEntity(const EntityPtr &entity, const ComponentPtr &comp
  */
 bool isStandardUnitName(const std::string &name);
 
+bool isStandardUnit(const UnitsPtr &units);
```
Comment (agarny) [inline src/utilities.h:404]: Need some documentation for this new method.

File: tests/printer/printer.cpp [hunk 4]
```diff
@@ -235,3 +235,20 @@ TEST(Printer, printModelWithTabs)
...
 }
+
+TEST(Printer, printModelWithStandardUnitsAdded)
+{
+    const std::string e =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"my_model\"/>\n";
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create("second");
+    libcellml::ModelPtr model = libcellml::Model::create("my_model");
```
Comment (agarny) [inline tests/printer/printer.cpp:245]: Minor: might want to swap those two lines? (Just feels "weird" to create a units before a model.)

**Review by agarny — CHANGES_REQUESTED** [review]
Looks fine to me pensing some minor issues.

**Review by agarny — APPROVED** [review]

**Review by nickerso — APPROVED** [review]

**Pull request merged**